src/Dapr.Actors/Client/ActorProxyFactory.cs

```diff
@@ -15,22 +16,27 @@ namespace Dapr.Actors.Client
     public class ActorProxyFactory : IActorProxyFactory
     {
         private readonly IDaprInteractor daprInteractor;
-        private ActorProxyOptions defaultOptions = new ActorProxyOptions();
+        private ActorProxyOptions defaultOptions;
 
         /// <inheritdoc/>
         public ActorProxyOptions DefaultOptions
         {
             get => this.defaultOptions;
-            set => this.defaultOptions = value ??
+            set
+            {
+                this.defaultOptions = value ??
                     throw new ArgumentNullException(nameof(DefaultOptions), $"{nameof(ActorProxyFactory)}.{nameof(DefaultOptions)} cannot be null");
+                this.daprInteractor.SetDaprApiToken(this.defaultOptions.DaprApiToken);
+            }
         }
 
         /// <summary>
         /// Initializes a new instance of the <see cref="ActorProxyFactory"/> class.
         /// </summary>
-        public ActorProxyFactory()
+        public ActorProxyFactory(ActorProxyOptions options = null)
         {
-            this.daprInteractor = new DaprHttpInteractor();
+            this.defaultOptions = options ?? new ActorProxyOptions();
+            this.daprInteractor = new DaprHttpInteractor(innerHandler:options?.HttpClientMessageHandler, apiToken:options?.DaprApiToken);
```

Comment:
So this introduces a bunch of problems, having a class that that's mutable, and shared, but allow has per-proxy state. The http interactor is shared (one per factory), but has it's settings modified per-proxy you create (line 51).

For instance if you called `Create` twice with two different values of the API token then it would crosstalk. 

My thoughts on how you should address this design-wise:
- Get rid of `options.HttpClientMessageHandler` - this addition breaks the design.
- If you need to make the message handler customizable for tests, make it possible pass in in constructor of `ActorProxyFactory` - otherwise fallback to a **static** instance on `ActorProxyFactory`.
- Create new instance of `DaprHttpInteractor` inside `Create` but **do not** store them